source/Calamari.Shared/Integration/Packages/NuGet/NuGetV3Downloader.cs

```diff
@@ -1,29 +1,96 @@
-﻿// Much of this class was based on code from https://github.com/NuGet/NuGet.Client. It was ported, as the NuGet libraries are .NET 4.5 and Calamari is .NET 4.0
+// Much of this class was based on code from https://github.com/NuGet/NuGet.Client. It was ported, as the NuGet libraries are .NET 4.5 and Calamari is .NET 4.0
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.//
 #if USE_NUGET_V2_LIBS
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
 using System.Net;
 using System.Net.Http;
+using System.Web;
 using Calamari.Common.Commands;
 using Calamari.Common.Plumbing.Logging;
+using Calamari.Deployment;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
+using Octopus.CoreUtilities.Extensions;
 using Octopus.Versioning;
+using Octopus.Versioning.Semver;
 
 namespace Calamari.Integration.Packages.NuGet
 {
     internal static class NuGetV3Downloader
     {
+        public static bool CanHandle(Uri feedUri, ICredentials feedCredentials, TimeSpan httpTimeout)
+        {
+            if (feedUri.ToString().EndsWith(".json", StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            return IsJsonEndpoint(feedUri, feedCredentials, httpTimeout);
+        }
+
+        static bool IsJsonEndpoint(Uri feedUri, ICredentials feedCredentials, TimeSpan httpTimeout)
+        {
+#if NET40
+            var request = WebRequest.Create(feedUri);
```

Comment:
I assume this `WebRequest` class is a utility class written by us? If so

1. Are there any Proxy settings that need to be respected for outbound requests from Calamari?
2. Does this class respect them?

Reply:
(I'm assuming, having read all the stuff in the rest of this class, that my questions are probably redundant)

Reply:
Nah WebRequest is the old way of doing things in .NET 4. I am pretty sure Andrew and I already confirmed proxy stuff was ok.